ako-infra/ingestion/avi_infra_controller.go

```diff
@@ -73,6 +77,137 @@ func (a *AviControllerInfra) VerifyAviControllerLicense() error {
+	err, _, segUuid := a.DeriveCloudNameAndSEGroupTmpl(tz)
+	if err == nil {
+		if segUuid == "" {
+			// The cloud does not have a SEG template set, use `Default-Group`
+			uri := "/api/serviceenginegroup/" + "?name=Default-Group"
+			result, err := lib.AviGetCollectionRaw(a.AviRestClients.AviClient[0], uri)
+			if err != nil {
+				utils.AviLog.Errorf("Get uri %v returned err %v", uri, err)
+				return false
+			}
+			// Construct an SE group based on parameters in the `Default-Group`
+			elems := make([]json.RawMessage, result.Count)
+			err = json.Unmarshal(result.Results, &elems)
+			if err != nil {
+				utils.AviLog.Errorf("Failed to unmarshal data, err: %v", err)
+				return false
+			}
+
+			for _, elem := range elems {
+				seg := models.ServiceEngineGroup{}
+				if err := json.Unmarshal(elem, &seg); err != nil {
+					utils.AviLog.Warnf("Failed to unmarshal data, err: %v", err)
+					continue
+				}
+
+				if !ConfigureSeGroup(a.AviRestClients.AviClient[0], &seg) {
+					return false
+				}
+			}
+
+		} else {
+			// se group template exists, use the same to fetch the SE group details and use it to create the new SE group
+			// The cloud does not have a SEG template set, use `Default-Group`
+			uri := "/api/serviceenginegroup/" + segUuid
+			result, err := lib.AviGetCollectionRaw(a.AviRestClients.AviClient[0], uri)
+			if err != nil {
+				utils.AviLog.Errorf("Get uri %v returned err %v", uri, err)
+				return false
+			}
+			// Construct an SE group based on parameters in the `Default-Group`
+			elems := make([]json.RawMessage, result.Count)
+			err = json.Unmarshal(result.Results, &elems)
+			if err != nil {
+				utils.AviLog.Errorf("Failed to unmarshal data, err: %v", err)
+				return false
+			}
+
+			for _, elem := range elems {
+				seg := models.ServiceEngineGroup{}
+				if err := json.Unmarshal(elem, &seg); err != nil {
+					utils.AviLog.Warnf("Failed to unmarshal data, err: %v", err)
+					continue
+				}
+
+				if !ConfigureSeGroup(a.AviRestClients.AviClient[0], &seg) {
+					return false
+				}
+			}
+		}
+	}
+	return true
+}
+
+// ConfigureSeGroup creates the SE group with the supplied properties, alters just the SE group name and the labels.
+func ConfigureSeGroup(client *clients.AviClient, seGroup *models.ServiceEngineGroup) bool {
+	// Change the name of the SE group
+	*seGroup.Name = lib.GetClusterID()
+
+	uri := "/api/serviceenginegroup/"
+	// Add the labels.
+	seGroup.Labels = lib.GetLabels()
+	response := models.ServiceEngineGroupAPIResponse{}
+	// If tenants per cluster is enabled then the X-Avi-Tenant needs to be set to admin for vrfcontext and segroup updates
+	if lib.GetTenantsPerCluster() && lib.IsCloudInAdminTenant {
+		SetAdminTenant := session.SetTenant(lib.GetAdminTenant())
+		SetTenant := session.SetTenant(lib.GetTenant())
+		SetAdminTenant(client.AviSession)
+		defer SetTenant(client.AviSession)
+	}
+
+	err := lib.AviPost(client, uri, seGroup, response)
```

Comment:
One basic doubt: We are either using "default SE group" or new SE Group from template. Do we need to always do post operation (as in "Default SE group" case , as group already exist, post might fail)

Reply:
Good question. There's a never an update expected on the SEG, so I missed a method on bootup on this PR that will check if the SE group exists or not with the right marker. If it does, then it will no-op. I wanted to add it post the integration PR.